packages/lodestar-utils/src/json.ts

```diff
@@ -0,0 +1,38 @@
+import {Json, toHexString} from "@chainsafe/ssz";
+
+export function defaultErrorToJson(obj: Error): Json {
+  const errObj: Json = {message: obj.message};
+  if (obj.stack) errObj.stack = obj.stack;
+  return errObj;
+}
+
+interface ICustomError extends Error {
+  toJson?: () => Json;
+}
+
+interface IToJson {
+  toJson: () => Json;
+}
+
+export function toJson(obj: unknown): Json {
+  if (obj == null) {
+    return null;
+  } else if (typeof obj === "number" || typeof obj === "string") {
+    return obj;
+  } else if (typeof obj === "bigint") {
+    return String(obj);
+  } else if (obj instanceof Uint8Array) {
+    return toHexString(obj);
+  } else if (obj instanceof Error) {
+    return (obj as ICustomError).toJson?.() ?? defaultErrorToJson(obj);
+  } else if (Array.isArray(obj)) {
+    return obj.map(toJson);
+  } else if ((obj as Partial<IToJson>).toJson) {
+    return (obj as IToJson).toJson();
+  } else if (typeof obj === "object") {
+    const jsonObj: Json = {};
+    Object.entries(obj as object).forEach(([k, v]) => (jsonObj[k] = toJson(v)));
+    return jsonObj;
+  }
+  throw new Error("Unable to convert unknown to json: " + obj);
```

Comment:
You are forgetting "boolean" and "functions" types, which will throw and error and are frequent types. I strongly argue against ever throwing an error here.
```ts
function toJson(arg: unknown): Json {
  switch (typeof arg) {
    case "bigint":
    case "symbol":
    case "function":
      return arg.toString();

    case "object":
      if (arg === null) return "null";
      if (Array.isArray(arg)) return arg.map(toJson);
      if (arg instanceof Uint8Array) return toHexString(arg);
      if (arg instanceof LodestarError) return arg.toJson();
      return lodash.mapValues(arg, (value) => toJson(value));

    // Already valid JSON
    case "number":
    case "string":
    case "undefined":
    case "boolean":
    default:
      return arg;
  }
}
```